vscode-trace-extension/src/utils/trace-explorer-resource-type-handler.ts

```diff
@@ -0,0 +1,82 @@
+/***************************************************************************************
+ * Copyright (c) 2024 BlackBerry Limited and others.
+ *
+ * Licensed under the MIT license. See LICENSE file in the project root for details.
+ ***************************************************************************************/
+import * as vscode from 'vscode';
+
+export type ResourceType = 'File' | 'Folder';
+export type ResourceTypeQuickPickItem = vscode.QuickPickItem & { type: ResourceType };
+
+export class TraceExplorerResourceTypeHandler {
+    private static instance: TraceExplorerResourceTypeHandler;
+    private doHandleFiles = true;
+    private doHandleFolders = true;
+    private quickpickItems: ResourceTypeQuickPickItem[] = [
+        { label: 'File', type: 'File' },
+        { label: 'Folder', type: 'Folder' }
+    ];
+
+    private constructor() {}
```

Comment:
Getting ESLint "no-empty-function" error. You can resolve this by adding an empty comment on purpose

` private constructor() { /** Empty constructor */ }`

Reply:
Ok, fixed!